jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/bpmn2/BPMN2DataServiceImpl.java

```diff
@@ -132,14 +206,113 @@ public ProcessDefinition buildProcessDefinition(String deploymentId,String bpmn2
         } finally {
         	BPMN2ProcessFactory.setBPMN2ProcessProvider(originalProvider);
             BPMN2DataServiceSemanticModule.dispose();
+            // BPMN2DataServiceExtensionSemanticModule.dispose() <-- not needed because the modules share the same ThreadLocal instance
```

Comment:
so we don't need to remove anything from thread local any more? It can be safely reused, right